oktapam/provider_factory.go

```diff
@@ -0,0 +1,53 @@
+package oktapam
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-mux/tf6muxserver"
+)
+
+func V6ProviderServerFactory(ctx context.Context) (func() tfprotov6.ProviderServer, error) {
+	v5Provider := Provider()
+	v6Provider := New()()
+	// SDKV2 used for tf plugin development is designed for maintaining tf plugins that are compatible with Plugin
+	// Protocol version 5. Plugins need to communicate with Terraform CLI, protocol version 5 is supported by CLI version
+	// 0.12 and later. Protocol version 6 support tf cli version 1.0 or later.
+
+	// To start using new TF Plugin Framework(https://developer.hashicorp.com/terraform/plugin/framework) we have two options -
```

Comment:
these comments lay out some options, but do not specify what we chose and why - i can figure it out from the code below, but i am not sure if other devs that are unfamiliar with TF would understand it.

Reply:
updated comment to indicate we decided to use the new terraform provider framework and which option we adopted here. For details how to add new resouce/data resource, I added the session in https://github.com/okta/terraform-provider-oktapam/blob/2923d112dc55b414dce55662e8de22fc1df4dfcb/README.md#L72